rclcpp/include/rclcpp/strategies/single_msg_memory_strategy.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_RCLCPP_BOUNDED_MSG_MEMORY_STRATEGY_HPP_
+#define RCLCPP_RCLCPP_BOUNDED_MSG_MEMORY_STRATEGY_HPP_
+
+#include <rclcpp/msg_memory_strategy.hpp>
+#include <rclcpp/macros.hpp>
+
+namespace rclcpp
+{
+namespace single_msg_memory_strategy
+{
+
+template<typename MessageT>
+class SingleMessageMemoryStrategy :
+  public msg_memory_strategy::MessageMemoryStrategy<MessageT>
+{
+public:
+  RCLCPP_MAKE_SHARED_DEFINITIONS(SingleMessageMemoryStrategy);
+  SingleMessageMemoryStrategy()
+  {
+    //typename std::enable_if<has_bounded_size<MessageT>::value, MessageT>::type message;
+    MessageT message;
+    message_ = std::make_shared<MessageT>(message);
+  }
+
+  std::shared_ptr<void> borrow_message()
+  {
+    return message_;
+  }
+
+  virtual void return_message(std::shared_ptr<void> &/*msg*/)
+  {
+  }
```

Comment:
Do you suggest an alternative? Doing a ~T() and then T() to "reset" the message isn't very efficient since it will likely cause reallocation of strings and vectors in the messages. Requiring the user to reset the message before returning it isn't a good idea. I don't see a better way to achieve the reuse of a message than to have the generator generate a reset function. For type masquerading we can make it optional, doing an explicit ~T() then T() if the type doesn't have a reset implementation.